recipes/python/lxml.yaml

```diff
@@ -0,0 +1,58 @@
+inherit: [cpackage, python3-cext-pkg, strip, install]
+
+metaEnvironment:
+    PKG_LICENSE: "BSD-3-Clause"
+    PKG_VERSION: "5.3.1"
+
+depends:
+    - name: libs::libxml2-bin
+      tools:
+          target-toolchain: host-compat-toolchain
+      use: [tools]
+    - name: python::cython
+      tools:
+          target-toolchain: host-compat-toolchain
+
+    - libs::libiconv-dev
+    - libs::libxml2-dev
+    - libs::libxslt-dev
+    - libs::zlib-dev
+    - use: []
+      depends:
+          - libs::libiconv-tgt
+          - libs::libxml2-tgt
+          - libs::libxslt-tgt
+          - libs::zlib-tgt
+
+checkoutSCM:
+    scm: url
+    url: https://github.com/lxml/lxml/archive/refs/tags/lxml-${PKG_VERSION}.tar.gz
+    digestSHA1: f2c04d861f72d7119697194f6e46d3cdfd272e0c
+    stripComponents: 1
+
+buildTools: [libxslt-tools, libxml2-tools, target-toolchain]
+buildScript: |
+    mkdir -p __src__
+    rsync -a $1/ __src__/
+    STATIC=""
+    if [[ $(cpackageLibraryType) == static ]]; then
+       STATIC="--static"
+       PYTHON3_VERSION="$(python3 -c "import sysconfig; print(sysconfig.get_config_var('VERSION'))")"
+       export LXML_STATIC_INCLUDE_DIRS="${BOB_DEP_PATHS[libs::libxslt-dev]}/usr/include"
+       export LXML_STATIC_INCLUDE_DIRS+=":${BOB_DEP_PATHS[libs::libxml2-dev]}/usr/include/libxml2"
+       export LXML_STATIC_INCLUDE_DIRS+=":${BOB_DEP_PATHS[libs::zlib-dev]}/usr/include"
+       export LXML_STATIC_INCLUDE_DIRS+=":${BOB_TOOL_PATHS['python3']}/../include/python${PYTHON3_VERSION}"
+       export LXML_STATIC_LIBRARY_DIRS="${BOB_DEP_PATHS[libs::libxslt-dev]}/usr/lib"
+       export LXML_STATIC_LIBRARY_DIRS+=":${BOB_DEP_PATHS[libs::libxml2-dev]}/usr/lib"
+       export LXML_STATIC_LIBRARY_DIRS+=":${BOB_DEP_PATHS[libs::libiconv-dev]}/usr/lib"
+       export LXML_STATIC_LIBRARY_DIRS+=":${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib"
+
+       sed -i "s#standard_libs = \[#standard_libs = \['xslt', 'exslt', 'xml2', 'iconv', 'z'#g" __src__/setupinfo.py
+       export CFLAGS="-I${BOB_TOOL_PATHS['python3']}/../include/python${PYTHON3_VERSION} ${CFLAGS:-}"
+    fi
+    export STATIC_CFLAGS=${CFLAGS}
+
+    python3CExtBuildSetuptools $(pwd)/__src__ --config-setting=--build-option=${STATIC}
```

Comment:
This is way too complicated. It looks like that cross-compilation works fine. But host/static compilation misses to link libiconv. It appears that the additional libraries that are named by pkg-config are not used. I would rather apply a patch in the sources to work around this problem.